src/lib.rs

```diff
@@ -181,6 +181,24 @@ impl<'a> fmt::Display for Record<'a> {
     }
 }
 
+/// Write Record Extension on io::Write
+pub trait WriteRecordExt: io::Write {
+    
+    /// write a record to a io::Write implementor
+    ///
+    /// returns the length of the written record
+    fn write_record(&mut self, record: Record) -> Result<()>;
+}
+
+impl<T> WriteRecordExt for T where T: io::Write {
+
+    fn write_record(&mut self, record: Record) -> Result<()>{
+        self.write_all(record.as_ref());
```

Comment:
Result of this operation is ignored as mentioned by rustc, but it definitely shouldn't.
Also `Ok(())` on line 197 is redundant since `Write::write_all` returns `Result<(), io::Error>`.
`map_err(Into::into)` or `map_err(From::from)` should solve this.